### PR TITLE
Implement codegenDecl for js

### PR DESCRIPTION
**Pull request by alehander42**
I am not sure how to test the output with regex in discard.
(I don't know if names are deterministic)

### CODE REVIEW

File: compiler/jsgen.nim [hunk 1]
```diff
@@ -1567,10 +1567,17 @@ proc genVarInit(p: PProc, v: PSym, n: PNode) =
   var
     a: TCompRes
     s: Rope
+    varCode: string
+    returnType: Rope
```
Comment (Araq) [inline compiler/jsgen.nim]: This doesn't have to be a ``var``  and also not a ``let``, you can inline the ``~""``  expression.

File: compiler/jsgen.nim [hunk 2]
```diff
@@ -2162,8 +2169,23 @@ proc genProc(oldProc: PProc, prc: PSym): Rope =
       returnStmt = "return $#;$n" % [a.res]
 
   p.nested: genStmt(p, prc.getBody)
-  let def = "function $#($#) {$n$#$#$#$#$#" %
-            [name, header,
+
+  var def: Rope
+  if not prc.constraint.isNil:
+    echo (prc.constraint.strVal & " {$n$#$#$#$#$#")
```
Comment (Araq) [inline compiler/jsgen.nim]: debug output

File: compiler/jsgen.nim (hunk 1 above)
Comment (alehander42) [reply compiler/jsgen.nim]: Yeah, I wanted to name it for clarity, because otherwise people might wonder why the ~"" is there

Comment (Araq) [reply compiler/jsgen.nim]: template returnType(): untyped = ~""

Comment (alehander42) [reply compiler/jsgen.nim]: Ok, fixed in last commit

**Pull request merged**